scenarios/python/console/captioning/captioning.py

```diff
@@ -0,0 +1,248 @@
+        # result is type SpeechRecognitionResult, which inherits the reason property from RecognitionResult. See:
+        # https://docs.microsoft.com/python/api/azure-cognitiveservices-speech/azure.cognitiveservices.speech.speechrecognitionresult
+        # https://docs.microsoft.com/python/api/azure-cognitiveservices-speech/azure.cognitiveservices.speech.recognitionresult
+        # e.result.reason is ResultReason.Canceled. To get the cancellation reason, see e.cancellation_details.reason.
+        if speechsdk.CancellationReason.EndOfStream == e.cancellation_details.reason :
+            helper.write_to_console(text = "End of stream reached.{}".format(linesep), user_config = user_config)
+            done = True
+        elif speechsdk.CancellationReason.CancelledByUser == e.cancellation_details.reason :
+            helper.write_to_console(text = "User canceled request.{}".format(linesep), user_config = user_config)
+            done = True
+        elif speechsdk.CancellationReason.Error == e.cancellation_details.reason :
+            # Error output should not be suppressed, even if suppress output flag is set.
+            print("Encountered error. Cancellation details: {}{}".format(e.cancellation_details, linesep))
+            done = True
+        else :
+            print("Request was cancelled for an unrecognized reason. Cancellation details: {}{}".format(e.cancellation_details, linesep))
+            done = True
+
+    def stopped_handler(e : speechsdk.SessionEventArgs) :
+        nonlocal done
+        helper.write_to_console(text = "Session stopped.{}".format(linesep), user_config = user_config)
+        done = True
+
+    if user_config["show_recognizing_results"] :
+        speech_recognizer.recognizing.connect(recognizing_handler)
+    speech_recognizer.recognized.connect(recognized_handler)
+    speech_recognizer.session_stopped.connect(stopped_handler)
+    speech_recognizer.canceled.connect(canceled_handler)
+
+    speech_recognizer.start_continuous_recognition()
+
+    while not done :
+        sleep(5)
+    speech_recognizer.stop_continuous_recognition()
+
+    return
+
+usage = """Usage: python captioning.py [...]
+
+  CONNECTION
+    --key KEY                     Your Azure Speech service subscription key.
+    --region REGION               Your Azure Speech service region.
+                                  Examples: westus, eastus
+
+  LANGUAGE
+    --languages LANG1,LANG2       Enable language identification for specified languages.
+                                  Example: en-US,ja-JP
+
+  INPUT
+    --input FILE                  Input audio from file (default input is the microphone.)
+    --url URL                     Input audio from URL (default input is the microphone.)
+    --format FORMAT               Use compressed audio format.
+                                  Valid only with --file or --url.
+                                  If this is not specified, uncompressed format (wav) is assumed.
```

Comment:
What does it mean to assume "wav" when the default is "any"?

Reply:
Yeah that is confusing. If you don't use --format at all, it assumes .wav. If you do use --format but don't provide a value, it assumes "any". I'll try to clarify.